image_get/worker/test/test_worker.py

```diff
@@ -0,0 +1,219 @@
+                self.requests_last_second.popleft()
+        self.requests_last_second.append(time.time())
+
+    def update_load(self):
+        original_load = self.load
+        utilization =\
+            len(self.requests_last_second) / self.max_requests_per_second
+        if utilization <= 0.8:
+            self.load = self.Load.LOW
+        elif 0.8 < utilization < 1:
+            self.load = self.Load.HIGH
+        else:
+            self.load = self.Load.OVERLOADED
+            if self.fail_if_overloaded:
+                assert False, "You DDoS'd the server!"
+        if self.load != original_load:
+            log.debug(f'Changed simulator load status to {self.load}')
+
+    def lag(self):
+        """ Determine how long a request should lag based on load. """
+        if self.load == self.Load.LOW:
+            wait = random.uniform(0.05, 0.15)
+        elif self.load == self.Load.HIGH:
+            wait = random.uniform(0.15, 0.6)
+        # Overloaded
+        else:
+            wait = random.uniform(2, 10)
+        return wait
+
+    async def get(self, url):
+        self.record_request()
+        self.update_load()
+        await asyncio.sleep(self.lag())
+        if self.load == self.Load.HIGH:
+            status = random.choice(self.high_load_status_choices)
+        elif self.load == self.Load.OVERLOADED:
+            status = random.choice(self.overloaded_status_choices)
+        else:
+            status = 200
+        return FakeImageResponse(status)
+
+
+def test_poll():
+    """ Test message polling and parsing."""
+    consumer = FakeConsumer()
+    msgs = [
+        {
+            'url': 'http://example.org',
+            'uuid': 'c29b3ccc-ff8e-4c66-a2d2-d9fc886872ca'
+        },
+        {
+            'url': 'https://creativecommons.org/fake.jpg',
+            'uuid': '4bbfe191-1cca-4b9e-aff0-1d3044ef3f2d'
+        }
+    ]
+    encoded_msgs = [json.dumps(msg) for msg in msgs]
+    for msg in encoded_msgs:
+        consumer.insert(msg)
+    res = poll_consumer(consumer=consumer, batch_size=2)
+    assert len(res) == 2
+
+
+def validate_thumbnail(img, identifier):
+    """ Check that the image was resized. """
+    i = Image.open(img)
+    width, height = i.size
+    assert width <= 640 and height <= 480
+
+
+@pytest.mark.asyncio
+async def test_pipeline():
+    """ Test that the image processor completes with a fake image. """
+    # validate_thumbnail callback performs the actual assertions
+    await process_image(
+        persister=validate_thumbnail,
+        session=FakeAioSession(),
+        url='fake_url',
+        identifier='4bbfe191-1cca-4b9e-aff0-1d3044ef3f2d'
+    )
+
+
+@pytest.mark.asyncio
+async def test_handles_corrupt_images_gracefully():
+    await process_image(
+        persister=validate_thumbnail,
+        session=FakeAioSession(corrupt=True),
+        url='fake_url',
+        identifier='4bbfe191-1cca-4b9e-aff0-1d3044ef3f2d'
```

Comment:
worker_1     | Traceback (most recent call last):
worker_1     |   File "worker/consumer.py", line 1, in <module>
worker_1     |     import worker.settings as settings
worker_1     | ModuleNotFoundError: No module named 'worker'
image_get_worker_1 exited with code 1